src/Microsoft.Health.Fhir.Core/Features/Definition/SearchParameterDefinitionManager.cs

```diff
@@ -107,5 +113,35 @@ public ValueSets.SearchParamType GetSearchParameterType(SearchParameterInfo sear
 
             return componentSearchParameter.Type;
         }
+
+        public void UpdateSearchParameterHashMap(Dictionary<string, string> updatedSearchParamHashMap)
+        {
```

Comment:
Should there be a way to remove a hash mapping? Currently this method gives a way to add or update a mapping, but not remove one.

Reply:
The underlying dictionary represents a resourceType - searchParameterHash mapping. Currently, I don't see a scenario where we will be removing supported resource types. Hence I have added only Get and Update methods.